src/domains/contact/components/UpdateContact/UpdateContact.tsx

```diff
@@ -22,18 +24,35 @@ export const UpdateContact = ({
 	onCancel,
 	onDelete,
 	onSave,
+	profileId,
 }: UpdateContactProps) => {
 	const { t } = useTranslation();
+	const env = useEnvironment();
+
+	const handleSave = async ({ name }: any) => {
+		const profile = env?.profiles().findById(profileId);
+		profile?.contacts().update(contact.id, { name });
```

Comment:
@faustbrian if I use an address array in `update` e.g
```js
await profile?.contacts().update(contact.id, {
	name,
	addresses: [ { name: "test", network: "devnet", address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT", coin: "ARK" }, ],
});
``` 
My tests fail with unrelated error, 
```
/home/goga/dev/ark/desktop-wallet/node_modules/react-dom/cjs/react-dom.development.js:154
      var evt = document.createEvent('Event'); // Keeps track of whether the user-provided callback threw an error. We
                         ^

TypeError: Cannot read property 'createEvent' of null
```

Not sure what I am missing here. Is there any example with setting addresses in contacts?

Reply:
`update` is async and sends a network request so you will need to mock and await. The react-dom error is probably just a side-effect of that.

Reply:
@goga-m had a quick look and it seems like there is no wallet with the coin and network you are using for the contact so it will fail to look up the data that is associated with the contact address. You will first need to create a wallet for the coin ARK with network event and then create the contact.

Reply:
Yeah, makes sense. Will refactor the code and create wallets with valid coin/wallet to feed the address update. Thanks!